llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -5229,7 +5229,7 @@ void SelectionDAGBuilder::visitTargetIntrinsic(const CallInst &I,
   // definition.
   const Function *F = I.getCalledFunction();
   bool HasChain = !F->doesNotAccessMemory();
```

Comment:
We should probably also have `|| !F->willReturn()` here. If it's a readnone function that may diverge, it also needs a chain, right? This would probably also help identify intrinsics that are incorrectly missing the attribute.

Reply:
This seems like a good idea. When I add that there are many test failures, and the few I looked into are resolved by adding WillReturn. If it's okay, I would prefer that be added and the relevant intrinsics fixed in a separate PR so that we can land this one sooner.